### PR TITLE
chore(Button): Convert examples to typescript

### LINKED ISSUE #6793: Button: Convert examples to typescript

### PR DESCRIPTION
**Pull request by jeffpuzzo**
**What**: Closes #6793

### CODE REVIEW

File: packages/react-core/src/components/Button/examples/ButtonProgress.tsx [hunk 1]
```diff
@@ -0,0 +1,29 @@
...
+  const [isPrimaryLoading, setIsPrimaryLoading] = React.useState(true);
+  const [isSecondaryLoading, setIsSecondaryLoading] = React.useState(true);
+
+  return (
+    <Flex>
+      <Button
+        isLoading={isPrimaryLoading}
+        variant="primary"
+        onClick={() => setIsPrimaryLoading(!isPrimaryLoading)}
+        {...(isPrimaryLoading && { spinnerAriaValueText: 'Loading' })}
```
Comment (nicolethoen) [inline packages/react-core/src/components/Button/examples/ButtonProgress.tsx:14]: This line is causing a problem when I hit the `ts>>js` button...

Comment (jeffpuzzo) [reply packages/react-core/src/components/Button/examples/ButtonProgress.tsx:14]: The mechanism used to convert between TS and JS seems to break when spreading props (at least for Button). Can you point me to where we do the conversion so I can take a look?

Comment (nicolethoen) [reply packages/react-core/src/components/Button/examples/ButtonProgress.tsx:14]: with about 70% confidence, I think that is what this line is supposed to do?
https://github.com/patternfly/patternfly-org/blob/main/packages/theme-patternfly-org/scripts/md/mdx-hast-to-jsx.js#L166

**Review by nicolethoen — APPROVED** [review]
Works great

issues with typescript -> javascript parsing will be resolved when[ this org PR](https://github.com/patternfly/patternfly-org/pull/2816) merges. This PR does not need to be blocked.

**Review by wise-king-sullyman — APPROVED** [review]

**Pull request merged**